### PR TITLE
Validate dnceng-linux-external-temp-staging queue

### PR DESCRIPTION
**Pull request by chcosta**
@jaredpar, I don't think I have enough permissions to the Roslyn repo to authorize this agent queue in Azure DevOps.  I may need some assistance getting this validated.

### CODE REVIEW

File: .vsts-dnceng.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 phases:
 - phase: Windows_Desktop_Unit_Tests
   queue:
-    name: Helix
+    name: dnceng-linux-external-temp-staging
```
Comment (jaredpar) [inline .vsts-dnceng.yml]: These are Windows jobs and the new pool is a linux one. Seems like something is wrong here.

Comment (chcosta) [reply .vsts-dnceng.yml]: hmm... good point.  copy-paste error

**Comment by jaredpar** [comment]
@chcosta I believe it's authorized now. Set it as the pool, saved and reverted.

**Comment by chcosta** [comment]
Hmm... sadly, I think it's more complicated than that.  If the build definition doesn't find the queue in the "default" branch, then it won't authorize it.  We have a bit of a quandary here.  

- I don't have permissions to push my changes to dotnet/Roslyn, 
- I had to push to chcosta/Roslyn.  I don't think you have access to my fork (but I can change that).
- You have to change the default branch to the branch with the yaml that has the queue, so, either my changes need to go to a branch of dotnet/Roslyn, or you have to change the build repo to chcosta/Roslyn instead of dotnet/Roslyn and set the branch.
- After that silliness is done, you can set the default queue to dotnet-external-temp-staging so it will authorize the queues in the specified default repo / branch / queue
- then undo it all.

Or, you can make the change and push it to dotnet/Roslyn and do the authorization thing.  We can IM if all that doesn't make sense.

**Comment by chcosta** [comment]
The build highlighted that the agents were running on 2 core machines, I'm creating new agents which run on 4 cores and hope to re-validate this queue today.

**Comment by chcosta** [comment]
I have the job running on an Azure "Standard D4_v3" machine, and it appears that it is going to timeout.  @jaredpar are there other special requirements for Roslyn public CI builds?

https://dnceng.visualstudio.com/public/_build/results?buildId=26291&view=logs

@alexperovich , how are these machines different from the ones provided by the Helix pool?

**Comment by jaredpar** [comment]
@chcosta 

> I have the job running on an Azure "Standard D4_v3" machine, and it appears that it is going to timeout.

How sure are you about this? The output of our build strongly suggest this is a dual core machine:

>  3 running, 53 queued,  0 completed

That "3 running" is always calculated by "1.5 * processor count".

**Comment by chcosta** [comment]
Nice.  Indeed.  The external staging machine doesn't appear to have been updated to D4_v3, it's still D2_v3.  It looks like the internal staging queue is D4_v3, and the prod queues (internal and external) is D4_v3.  

Prod is what we really care about now that the agents are available.  Can you go through the hassle of authorizing the `dotnet-external-temp` queue so that we can test that queue out?  We have 50 machines there but I wouldn't suggest you take a hard dependency on that queue for your master branch runs yet as they are slightly limited on disk drive space (we're working on fixing this).

**Comment by chcosta** [comment]
FTR, I don't know why external staging didn't get upgraded but the other queues did.  I think it's just some nuanced magic of how Azure handles ARM deployments / scalesets.

**Pull request closed without merging**